parametric/test_headers_tracestate_dd.py

```diff
@@ -0,0 +1,471 @@
+        )
+
+        # 2) x-datadog-origin is NOT a well-known value
+        headers2 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "7890123456789012"],
+                ["x-datadog-parent-id", "1234567890123456"],
+                ["x-datadog-origin", "tracing2.0"],
+            ],
+        )
+
+        # 3) x-datadog-origin has invalid characters
+        headers3 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "7890123456789012"],
+                ["x-datadog-parent-id", "1234567890123456"],
+                ["x-datadog-origin", "synthetics;,=web"],
+            ],
+        )
+
+        # 4) tracestate[dd][o] is not present
+        headers4 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                ["tracestate", "foo=1,dd=s:-1"],
+            ],
+        )
+
+        # 5) tracestate[dd][o] is present and is a well-known value
+        headers5 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                ["tracestate", "foo=1,dd=s:-1;o:synthetics-browser"],
+            ],
+        )
+
+        # 6) tracestate[dd][o] is present and is NOT a well-known value
+        headers6 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                ["tracestate", "foo=1,dd=s:-1;o:tracing2.0"],
+            ],
+        )
+
+    # 1) x-datadog-origin is a well-known value
+    # Result: Origin set to header value
+    assert headers1["x-datadog-origin"] == "synthetics-browser"
+
+    traceparent1, tracestate1 = get_tracecontext(headers1)
+    dd_items1 = tracestate1["dd"].split(";")
+    assert "traceparent" in headers1
+    assert "tracestate" in headers1
+    assert "o:synthetics-browser" in dd_items1
+
+    # 2) x-datadog-origin is NOT a well-known value
+    # Result: Origin set to header value
+    assert headers2["x-datadog-origin"] == "tracing2.0"
+
+    traceparent2, tracestate2 = get_tracecontext(headers2)
+    dd_items2 = tracestate2["dd"].split(";")
+    assert "traceparent" in headers2
+    assert "tracestate" in headers2
+    assert "o:tracing2.0" in dd_items2
+
+    # 3) x-datadog-origin has invalid characters
+    # Result: Origin set to header value, where invalid characters replaced by '_'
+    assert headers3["x-datadog-origin"] == "synthetics;,=web"
+
+    traceparent3, tracestate3 = get_tracecontext(headers3)
+    dd_items3 = tracestate3["dd"].split(";")
+    assert "traceparent" in headers3
+    assert "tracestate" in headers3
+    assert "o:synthetics__~web" in dd_items3
```

Comment:
Wouldn't this be the correct encoding? Changing `=` to `~` rather than `_` ?

Reply:
Thanks for this fix, this makes sense to standardize string injection/extraction